painttiming.bs

```diff
@@ -62,9 +86,24 @@ Formally, we consider the user agent to have "rendered" a document when it has p
 
     NOTE: The rendering pipeline is very complex, and the timestamp should be the latest timestamp the user agent is able to note in this pipeline (best effort). Typically the time at which the frame is submitted to the OS for display is recommended for this API.
 
+<dfn export>Contentful</dfn>: An element counts as contentful when one or more of the following apply:
+* The element contains a [=text node=], with at least one character, excluding [=white-space=].
+* The element is a [=replaced image=] representing a [=loaded=] image.
+* The element's current style contains a [=loaded=] [=background-image=] or [=mask-image=].
+* The element is a [=canvas=] with its [=context mode=] set to any value other than <code>none</code>.
+* The element is an [=svg element with rendered descendants=].
+
+<dfn export>Paintable</dfn>: An element counts as paintable when all of the following apply:
+* The element intersects with the document, as defined by [=Intersection rect algorithm=]. 
+
+    NOTE: This covers the cases where the element is transformed to zero size, or is positioned in negative coordinates,
+
+* The element's [=visibility=] is visible.
+* The element's [=opacity=] is greater than 0.
```

Comment:
Here it may be worth adding a note that this is just covering some minor cases. I.e. a descendant of an element that is marked as hidden or has 0 opacity may be considered visible even though it won't be painted.

Reply:
You're right - it needs to be "The element's computed opacity, as well as the opacity of all of its decsendants, is greater than 0"